### DISCUSSION TITLE
Improving to First Run

### DISCUSSION THREAD

**Discussion by jamesvandyne** [jamesvandyne/tanzawa#78] [Ideas]
Once you get Tanzawa running, the first run experience is lacking. Eventually it'd be great to have a nice polished workflow, but I think just the basics should suffice for now.

There's two (or three) pain points:

1. Running Tanzawa in Development
2. Deploying Tanzawa
3. Initial setup

##  Running Tanzawa in Development

I don't think it's that _bad_ at the moment. Running a Docker container to standardize Python/Geo libraries is super helpful. Manually running the dev server from within the container is less than idea. Maybe this could be improved to always run the dev server on boot?

## Deploying Tanzawa

Deploying Python applications in general is a complex process. Providing a production Docker container (running gunicorn) and instructions for Ubuntu (systemd) is good for those running their own servers. Maybe provide 1-click deploys to Digital Ocean?  Can we lower the barrier to entry and do Heroku/Vercel/ other PaaS?

## Initial Setup

I've started on a PR  ( #74) to help address some of the pain points of initial setup. 

In the current install instructions, you register your account via the command line. That's not great. What I'd like to do is, if no users are registered to redirect to a First Run page / workflow. On this page users could input their username / email / blog name. Perhaps a second step modify Streams? After that's complete, perhaps a link to write your first post _or_ Import from Wordpress? ( Sidebar: Wordpress import exists at `/a/wordpress/upload/` but I've only tested it with my personal blog, so it may not work perfectly )

**Comment by leonstafford** [comment]
re 

> Any ideas for alternatives for the secret key gen? When I was researching how to do it, I ran into issues where it wouldn't run on macOS (BSD utils) or commands are Linux only (gnu utils) - Python seemed like a good compromise as it's standard on most platforms these days? Maybe this should move to a discussion...

Aside: [ShellCheck](https://github.com/koalaman/shellcheck) can be great for warning when using syntax not widely available. 

Not sure how secure that key needs to be, maybe can borrow from some other popular shell scripts, but can't think of what those would be right now.... Perl should be a bit more ubiquitous. Could do a architecture check/switch statement and use preferred built-in password generators? 

I dunno - I try to avoid security, datetime and language encodings in my software development life and somehow getting away with it :D

**Comment by pimoore** [comment]
I think the idea of having a deployment to Netlify/Vercel/Render et. al. would be very useful to those who’d rather not admin their own server, myself being one of them.

**Comment by rmdes** [comment]
I'm trying to package the current docker implementation into a cloudron package (docker based)
but there are several parts of the deploy that I'm not sure how to implement in the cloudron context, still exploring :)

**Comment by jamesvandyne** [comment]
Oh nice! Let me know if there’s anything I can answer that may help!
